src/counts.jl

```diff
@@ -37,7 +37,7 @@ end
 
 function addcounts!(r::AbstractArray, x::IntegerArray, levels::IntUnitRange, wv::WeightVec)
     k = length(levels)
-    length(r) == k || throw(DimensionMismatch())
+    length(r) == length(wv) == k || throw(DimensionMismatch())
```

Comment:
isn't this the wrong comparison?  the weights should have the same length indices as the `x` vector unless I'm misunderstanding this code

Reply:
```suggestion
    length(r) == k || throw(DimensionMismatch())
    length(wv) == length(x) || throw(DimensionMismatch())
```

Reply:
Or even better:
```suggestion
    length(r) == k ||
        throw(DimensionMismatch("length of levels ($k) must match length of destination array ($(length(r)))"))
    length(wv) == length(x) ||
        throw(DimensionMismatch("length of weights vector ($(length(wv))) must match length of data vector ($(length(x))"))
```